### PR TITLE
Registration/Login by Scandie

### PR DESCRIPTION
**Pull request by Scandie**

### CODE REVIEW

File: rate_counter/accounts/forms.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    username = forms.CharField()
+    password = forms.CharField(widget=forms.PasswordInput)
+
+    def clean(self):
+        username = self.cleaned_data.get("username")
+        password = self.cleaned_data.get("password")
+        if username and password:
+            user = authenticate(username=username, password=password)
+            if not user:
+                raise forms.ValidationError("User does not exist")
```
Comment (wolendranh) [inline rate_counter/accounts/forms.py]: I suggest change message to
`"User {} does not exist".format(username)`

File: rate_counter/accounts/forms.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        return password
+
+    def clean_email2(self):
+        email = self.cleaned_data.get('email')
+        email2 = self.cleaned_data.get('email2')
+        if email != email2:
+            raise forms.ValidationError("Emails must match")
+        email_qs = User.objects.filter(email=email)
+        if email_qs.exists():
+            raise forms.ValidationError("This email addres already exists")
```
Comment (wolendranh) [inline rate_counter/accounts/forms.py]: addres -> address

File: rate_counter/accounts/views.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+from django.shortcuts import render, redirect
+
+from .forms import UserLoginForm, UserRegisterForm
+# Create your views here.
+
+
+def login_view(request):
+    form = UserLoginForm(request.POST or None)
+    title = "Login"
+    next = request.GET.get('next')
```
Comment (wolendranh) [inline rate_counter/accounts/views.py]: I guess 'next' is reserved by python keyword

File: rate_counter/static/rate/css/base.css [hunk 4]
```diff
@@ -17,4 +17,19 @@ th {
 .chart_div {
     width: 900px;
     height: 500px;
-}
\ No newline at end of file
+}
+
+.col-sm-5 {
```
Comment (wolendranh) [inline rate_counter/static/rate/css/base.css]: here you override Bootstrap built in css class. It is not recommended to do.
Html elements work with multiple classes.
For example if you have:
`<div class="col-sm-5">`
and you want to change style of it I recommend to do it like
`<div class="col-sm-5 my_custom_class">`
so you will not brake Bootstrap styles.

File: rate_counter/templates/tables_list.html [hunk 5]
```diff
@@ -7,16 +7,27 @@
 {% endblock title %}
 
 {% block content %}
```
Comment (wolendranh) [inline rate_counter/templates/tables_list.html:9]: statistic and profile link should not be visible for not authorised user

File: rate_counter/templates/tables_list.html [hunk 6]
```diff
@@ -7,16 +7,27 @@
...
-    {% if object.user.get_full_name %}
-        <p> Author: {{ obj.user.get_full_name }}</p>
-    {% else %}
-        <p> Author: {{ object.user }}</p>
-    {% endif %}
-{% endfor %}
-    <br><br>
-    <p><a href="{% url 'rate:create'%}" class="btn btn-default" role="button">Create New Table</a></p>
+    <h1>
+        <p><a href="{% url 'login' %}"  role="link">Login</a></p>
```
Comment (wolendranh) [inline rate_counter/templates/tables_list.html:27]: Those links are displayed on a page in a "middle of nowhere" :)

If possible make some box for them

File: rate_counter/accounts/forms.py (hunk 1 above)
Comment (Scandie) [inline rate_counter/accounts/forms.py]: Replaced message to a suggested one.

File: rate_counter/accounts/forms.py (hunk 2 above)
Comment (Scandie) [inline rate_counter/accounts/forms.py]: Fixed.

File: rate_counter/accounts/views.py (hunk 3 above)
Comment (Scandie) [inline rate_counter/accounts/views.py]: Changed to 'next_url'.

File: rate_counter/static/rate/css/base.css (hunk 4 above)
Comment (Scandie) [inline rate_counter/static/rate/css/base.css]: Created my own class "log-reg-brd".

File: rate_counter/templates/tables_list.html (hunk 5 above)
Comment (Scandie) [inline rate_counter/templates/tables_list.html:9]: Added condition for that.

File: rate_counter/templates/tables_list.html (hunk 6 above)
Comment (Scandie) [inline rate_counter/templates/tables_list.html:27]: Isn't it enough that links are inside of div? Besides, it is a temporary decision, i would like to add something more interesting to a main page.

Comment (wolendranh) [inline rate_counter/templates/tables_list.html:27]: make sense. let's make it more 'attractive' in other tickets.

**Pull request merged**